gogh.sh

```diff
@@ -230,40 +277,137 @@ echo -e "\nThemes:\n"
 
 for TH in "${THEMES[@]}"; do
 
-    KEY=$(printf "%02d" $NUM)
-    FILENAME=${TH::$((${#TH}-3))}
-    FILENAME_SPACE=${FILENAME//-/ }
+  KEY=$(printf "%02d" $NUM)
+  FILENAME=${TH::$((${#TH}-3))}
+  FILENAME_SPACE=${FILENAME//-/ }
 
-    echo -e "    (\\033[0m\033[0;34m $KEY \\033[0m\033[0m) $(capitalize "${FILENAME_SPACE}")"
+  echo -e "    ($(tput setaf 4) $KEY $(tput sgr0)) $(capitalize "${FILENAME_SPACE}")"
 
-    ((NUM++))
+  ((NUM++))
 
 done
+echo -e "    ($(tput setaf 4) ALL $(tput sgr0)) All themes"
 
 # |
 # | ::::::: Select Option
 # |
-echo -e "\nUsage : Enter Desired Themes Numbers (\\033[0m\033[0;34mOPTIONS\\033[0m\033[0m) Separated By A Blank Space"
-echo -e "        Press \033[0;34mENTER\\033[0m without options to Exit\n"
-read -p 'Enter OPTION(S) : ' -a OPTION
+echo -e "\nUsage : Enter Desired Themes Numbers ($(tput setaf 4)OPTIONS$(tput sgr0)) Separated By A Blank Space"
+echo -e "        Press $(tput setaf 4)ENTER$(tput sgr0) without options to Exit\n"
+read -r -p 'Enter OPTION(S) : ' -a OPTION
+
+# Automagically generate options if user opts for all themes
+[[ "$OPTION" == ALL ]] && OPTION=($(seq -s " " $ARRAYLENGTH))
+
+# |
+# | ::::::: Get terminal
+# |
+if [[ -z "${TERMINAL:-}" ]]; then
+  # |
+  # | Check for the terminal name (depening on os)
+  # | ===========================================
+  OS="$(uname)"
+  if [[ "$OS" = "Darwin" ]]; then
+    TERMINAL=$TERM_PROGRAM
+  elif [[ "${OS#CYGWIN}" != "${OS}" ]]; then
+    TERMINAL="mintty"
+  else
+    # |
+    # | Depending on how the script was invoked, we need
+    # | to loop until pid is no longer a subshell
+    # | ===========================================
+    pid="$$"
+    TERMINAL="$(ps -h -o comm -p $pid)"
+    while [[ "${TERMINAL:(-2)}" == "sh" ]]; do
+      pid="$(ps -h -o ppid -p $pid)"
+      TERMINAL="$(ps -h -o comm -p $pid)"
+    done
+  fi
+fi
+
+
+# |
+# | ::::::: Fancy progressbar for lengthy operations
+# |
+if [[ ${#OPTION[@]} -gt 5 ]]; then
+  if [[ "$(uname)" = "Darwin" ]]; then
+    eval "$(curl -sLo- https://git.io/progressbar)" 2> /dev/null
```

Comment:
Link to a specific release of your library, not master.

And use raw GitHub links directly, not through `git.io`. If you don't want the line to be long, store the URL in a variable.

The users of this repo need to know specific code they are running on their computer.